tree/dataframe/test/dataframe_vary.cxx

```
@@ -604,56 +602,227 @@ TEST_P(RDFVary, VariedHistosMustHaveNoDirectory)
 
-TEST_P(RDFVary, DISABLED_VaryDefinePerSample)
+TEST_P(RDFVary, DISABLED_VaryDefinePerSample) {}
+
+TEST(RDFVary, WithRange) // no Range in multithreaded runs
 {
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Range(5)
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Range(3)
+               .Sum<int>("x");
+   auto hs = VariationsFor(h);
 
+   EXPECT_EQ(*h, 3);
+   EXPECT_EQ(hs["nominal"], 3);
+   EXPECT_EQ(hs["x:0"], 0);
+   EXPECT_EQ(hs["x:1"], 6);
 }
 
-TEST_P(RDFVary, DISABLED_WithRange)
+TEST_P(RDFVary, VaryRedefine)
 {
+   // first redefine and then vary
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Redefine("x", [] { return 25; })
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Sum<int>("x");
+   auto hs = VariationsFor(h);
 
+   EXPECT_EQ(*h, 250);
+   EXPECT_EQ(hs["nominal"], 250);
+   EXPECT_EQ(hs["x:0"], 240);
+   EXPECT_EQ(hs["x:1"], 260);
 }
 
-TEST_P(RDFVary, DISABLED_VaryRedefine)
+TEST_P(RDFVary, RedefineVariedColumn)
 {
+   // first vary and then redefine
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Redefine("x", [] { return 25; })
+               .Sum<int>("x");
+   auto hs = VariationsFor(h);
 
+   EXPECT_EQ(*h, 250);
+   EXPECT_EQ(hs["nominal"], 250);
+   EXPECT_EQ(hs["x:0"], 35);
+   EXPECT_EQ(hs["x:1"], 55);
```

uhmmm I'm surprised this works, I think it's confusing that only the nominal value is overwritten. Redefine-ing a varied column should probably throw an exception instead. Please update the test, I'll update RDF :smile:

---

Just let me know if it should give logical or runtime error, so that I correct it accordingly.

---

runtime error

---

Test is still passing with latest master. My idea is to simply add a `TODO` remark (so that I do not commit a failing test).

---

nope, as I mentioned above, please update the test, i'll update RDF